code/modules/research/designs/guncrafting_designs.dm

```diff
@@ -0,0 +1,587 @@
+/datum/design/guncrafting_frame_energy
+	name = "Gun Frame: Energy"
+	desc = "No pin included, have fun."
+	id = "guncrafting_frame_energy"
+	req_tech = list("combat" = 1, "materials" = 1)
+	build_type = PROTOLATHE
+	materials = list(MAT_METAL = 500, MAT_GLASS = 300)
+	build_path = /obj/item/weapon/gun/energy/frame/testing
+	category = list("Guncrafting Parts")
+
+/datum/design/guncrafting_frame_energy_high
+	name = "Gun Frame: High Capacity Energy"
+	desc = "No pin included, have fun."
+	id = "guncrafting_frame_energy_high"
+	req_tech = list("powerstorage" = 2)
+	build_type = PROTOLATHE
+	materials = list(MAT_METAL = 700, MAT_GLASS = 300)
+	build_path = /obj/item/weapon/gun/energy/frame/testing/high
+	category = list("Guncrafting Parts")
+
+/datum/design/guncrafting_frame_energy_super
+	name = "Gun Frame: Super Energy"
+	desc = "No pin included, have fun."
+	id = "guncrafting_frame_energy_super"
+	req_tech = list("powerstorage" = 3, "materials" = 3)
+	build_type = PROTOLATHE
+	materials = list(MAT_METAL = 700, MAT_GLASS = 900)
+	build_path = /obj/item/weapon/gun/energy/frame/testing/super
+	category = list("Guncrafting Parts")
+
+/datum/design/guncrafting_frame_energy_hyper
+	name = "Gun Frame: Hyper Energy"
+	desc = "No pin included, have fun."
+	id = "guncrafting_frame_energy_hyper"
+	req_tech = list("powerstorage" = 5, "materials" = 5, "engineering" = 5)
+	build_type = PROTOLATHE
+	materials = list(MAT_METAL = 700, MAT_GLASS = 900, MAT_GOLD = 150, MAT_SILVER = 150, MAT_GLASS = 80)
+	build_path = /obj/item/weapon/gun/energy/frame/testing/hyper
+	category = list("Guncrafting Parts")
+
+/datum/design/guncrafting_frame_energy_bluespace
+	name = "Gun Frame: Bluespace Energy"
+	desc = "No pin included, have fun."
+	id = "guncrafting_frame_energy_bluespace"
+	req_tech = list("powerstorage" = 6, "materials" = 5, "engineering" = 5, "bluespace" = 5)
+	build_type = PROTOLATHE
+	materials = list(MAT_METAL = 800, MAT_GLASS = 900, MAT_GOLD = 120, MAT_SILVER = 150, MAT_GLASS = 160, MAT_DIAMOND = 160, MAT_TITANIUM = 300)
```

Comment:
Why am I getting yelled at for buffing science when goofball is allowed to make 40x capacity guns that take clickspamming/automatic fire for more then 2 full minutes to drain and let someone take three blobs down in a row solo?